### ISSUE TITLE
Epoch table missing row in some instances

### ISSUE THREAD

**Issue by rhyslbw** [input-output-hk/cardano-db-sync#111]
A running instance is missing an epoch row (194), which may have occurred at the epoch boundary. An instance built after this boundary did not have the same result.

**Comment by erikd** [comment]
This seems to happen at the epoch boundary. I added a test/debug feature in #110 that makes this easier to reproduce. Running the `cardano-db-sync-extended` with an added `--rollback-to-slot 4214000` will force an initial rollback to the very start of epoch 195. It will then roll forward and
```
select * from epoch order by no asc offset 190 ;
```
should show a missing row for epoch 194.

The problem is almost certainly in the logic here https://github.com/input-output-hk/cardano-db-sync/blob/master/cardano-db-sync/src/Cardano/DbSync/Plugin/Epoch.hs#L90-L107 which I tweaked just recently.
